scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2110,6 +2126,12 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     if (token.is[LF] && ahead { token.is[T] }) newLineOpt()
   }
 
+  def newLineOptWhenFollowedByIndentationAnd[T <: Token: TokenInfo]: Unit = {
+    val prev = currentIndentation
+    if (token.is[LF] && prev > 0 && dialect.allowSignificantIndentation &&
+      ahead { token.is[T] && currentIndentation > prev }) newLineOpt()
```

Comment:
does the formatter allow this? i thought ahead was the body..

Reply:
Not sure I understand? This parses and formats properly. I moved it out to a separate def anyway.